File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/attribute/fragments/presenter/Fragment2Presenter.java
package com.nerdscorner.mvplib.testapp.events.attribute.fragments.presenter;

import android.support.annotation.NonNull;

import com.nerdscorner.mvplib.events.presenter.BaseFragmentPresenter;
import com.nerdscorner.mvplib.testapp.events.attribute.fragments.model.Fragment2Model;
import com.nerdscorner.mvplib.testapp.events.attribute.fragments.view.Fragment2View;

import org.greenrobot.eventbus.NoSubscriberEvent;
import org.greenrobot.eventbus.Subscribe;

public class Fragment2Presenter extends BaseFragmentPresenter<Fragment2View, Fragment2Model> {
    public Fragment2Presenter(@NonNull Fragment2View view, @NonNull Fragment2Model model) {
        super(view, model);
        view.setText("Fragment 2");
    }

    @Subscribe
    public void onNoSubscribers(NoSubscriberEvent event) {
    }
}

File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/ui/mvp/presenter/Fragment2Presenter.kt
package com.nerdscorner.mvplib.testapp.events.ui.mvp.presenter

import com.nerdscorner.mvplib.events.bus.Bus
import com.nerdscorner.mvplib.events.presenter.BaseFragmentPresenter
import com.nerdscorner.mvplib.testapp.events.ui.mvp.model.Fragment2Model
import com.nerdscorner.mvplib.testapp.events.ui.mvp.view.Fragment2View

import org.greenrobot.eventbus.NoSubscriberEvent
import org.greenrobot.eventbus.Subscribe

class Fragment2Presenter(view: Fragment2View, model: Fragment2Model, bus: Bus) :
        BaseFragmentPresenter<Fragment2View, Fragment2Model>(view, model, bus) {
    init {
        view.setText("Fragment 2")
    }

    @Subscribe
    fun onNoSubscribers(event: NoSubscriberEvent) {
    }
}
